monkey/monkey_island/cc/services/authentication_service/flask_resources/agent_otp_login.py

```diff
@@ -17,31 +22,46 @@ class AgentOTPLogin(AbstractResource):
     A client may authenticate with the Island by providing a one-time password.
     """
 
-    urls = ["/api/agent-otp-login"]
+    urls = ["/api/agent-otp-login/<uuid:agent_id>"]
 
     def __init__(self, authentication_facade: AuthenticationFacade):
         self._authentication_facade = authentication_facade
 
-    def post(self):
+    @include_auth_token
+    def post(self, agent_id: AgentID):
         """
         Gets the one-time password from the request,
         and returns an authentication token and a refresh token
+        for a particular Agent
 
+        :param agent_id: The ID of the Agent trying to log in
         :return: Authentication token in the response body
         """
 
         try:
-            cred_dict = json.loads(request.data)
-            otp = cred_dict.get("otp", "")
+            otp = json.loads(request.data).get("otp", "")
             if self._validate_otp(otp):
-                refresh_token = "refreshtoken"
-
-                response = make_response(
-                    {"response": {"user": {ACCESS_TOKEN_KEY_NAME: "supersecrettoken"}}}
+                agent_user = register_user(
+                    RegisterForm(
+                        username=str(agent_id),
+                        password=secure_generate_random_string(32),
```

Comment:
Let's expand the character set to include special characters. By default this is only using alphanumerics.